File: venda/src/core/oop/produto-fisico.ts
import { ItemVenda } from "./item-venda.js";

export class ProdutoFisico extends ItemVenda {
  constructor(
    id: string,
    nome: string,
    preco: number,
    public peso: number
  ) {
    super(id, nome, preco);
  }

  getTipo(): string {
    return "Produto Físico";
  }
}

File: venda/projeto_vendas_1/servidor-java/ProdutoFisico.java

public class ProdutoFisico extends ItemVenda {
    private double peso;

    public ProdutoFisico(int id, String nome, double preco, double peso) {
        super(id, nome, preco);
        this.peso = peso;
    }

    public double getPeso() {
        return peso;
    }
}
